pkg/serverless/invocationlifecycle/lifecycle.go

```diff
@@ -230,21 +222,47 @@ func (lp *LifecycleProcessor) OnInvokeStart(startDetails *InvocationStartDetails
 		ev = event
 		lp.initFromLambdaFunctionURLEvent(event, region, account, resource)
 	case trigger.LegacyStepFunctionEvent:
-		var event events.StepFunctionEvent
+		var event events.StepFunctionEvent[events.StepFunctionPayload]
 		if err := json.Unmarshal(payloadBytes, &event); err != nil {
 			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
 			break
 		}
 		ev = event.Payload
-		lp.initFromStepFunctionPayload(event.Payload)
 	case trigger.StepFunctionEvent:
 		var eventPayload events.StepFunctionPayload
 		if err := json.Unmarshal(payloadBytes, &eventPayload); err != nil {
 			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
 			break
 		}
 		ev = eventPayload
-		lp.initFromStepFunctionPayload(eventPayload)
+	case trigger.LegacyNestedStepFunctionEvent:
+		var event events.StepFunctionEvent[events.NestedStepFunctionPayload]
+		if err := json.Unmarshal(payloadBytes, &event); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		ev = event.Payload
+	case trigger.NestedStepFunctionEvent:
+		var eventPayload events.NestedStepFunctionPayload
+		if err := json.Unmarshal(payloadBytes, &eventPayload); err != nil {
+			log.Debugf("Failed to unmarshal %s event: %s", stepFunction, err)
+			break
+		}
+		ev = eventPayload
+	case trigger.LegacyLambdaRootStepFunctionEvent:
+		var event events.StepFunctionEvent[events.LambdaRootStepFunctionPayload]
```

Comment:
It felt a little repetitive to repeat the whole case for every "legacy lambda" case when in reality the only difference is the whole payload is wrapped in a `"Payload": {...}`

But the alternative was to parse out this `Payload` somewhere upstream so we treat legacy vs non-legacy the same here. We're basically doing the same thing because the `extractors` won't know the difference. I also didn't want to modify the value as it's being passed down